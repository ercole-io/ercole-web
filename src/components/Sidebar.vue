<template>
  <sidebar-menu
    :menu="menu"
    :collapsed="true"
    width="250px"
    :hiddenOnCollapse="true"
    :showOneChild="true"
    @toggle-collapse="onToggleCollapse"
    data-sidebar
  >
    <span slot="toggle-icon">
      <b-icon pack="fas" icon="bars" size="is-small" />
    </span>
  </sidebar-menu>
</template>

<script>
import { SidebarMenu } from 'vue-sidebar-menu'

export default {
  components: {
    SidebarMenu
  },
  methods: {
    onToggleCollapse(collapsed) {
      this.$emit('collapsedSidebar', collapsed)
    }
  },
  computed: {
    menu() {
      return [
        {
          href: { name: 'dashboard' },
          title: this.$i18n.t('menu.dashboard'),
          icon: 'fas fa-home'
        },
        {
          href: { name: 'hosts' },
          title: this.$i18n.t('menu.hosts'),
          icon: 'fas fa-server'
        },
        {
          title: this.$i18n.t('menu.databases'),
          icon: 'fas fa-database',
          child: [
            {
              title: this.$i18n.t('menu.allTechs'),
              href: { path: '/databases' },
              icon: 'fas fa-cog'
            },
            {
              title: this.$i18n.t('menu.oracle'),
              icon: 'fas fa-cog',
              child: [
                {
                  title: this.$i18n.t('menu.dbList'),
                  href: { name: 'oracle' },
                  icon: 'fas fa-database'
                },
                {
                  href: { name: 'addm' },
                  title: this.$i18n.t('menu.addm'),
                  icon: 'fas fa-heartbeat'
                },
                {
                  href: { name: 'segment-advisor' },
                  title: this.$i18n.t('menu.segAdvisor'),
                  icon: 'fas fa-columns'
                },
                {
                  href: { name: 'patch-advisor' },
                  title: this.$i18n.t('menu.patAdvisor'),
                  icon: 'fas fa-band-aid'
                }
              ]
            },
            {
              title: this.$i18n.t('menu.mysql'),
              icon: 'fas fa-cog',
              child: [
                {
                  title: this.$i18n.t('menu.dbList'),
                  href: { name: 'mysql' },
                  icon: 'fas fa-database'
                }
              ]
            }
          ]
        },
        {
          href: { name: 'hypervisors' },
          title: this.$i18n.t('menu.hypervisors'),
          icon: 'fas fa-object-group'
        },
        // {
        //   href: { name: 'middleware' },
        //   title: 'Middlewares',
        //   icon: 'fas fa-globe'
        // },
        {
          href: { name: 'engineered-systems' },
          title: this.$i18n.t('menu.engSystems'),
          icon: 'far fa-calendar-times'
        },
        {
          href: { name: 'alerts' },
          title: this.$i18n.t('menu.alerts'),
          icon: 'fas fa-bell'
        },
        {
          title: this.$i18n.t('menu.licenses'),
          icon: 'fas fa-dollar-sign',
          child: [
            {
              href: { name: 'licenses-agreement' },
              title: this.$i18n.t('menu.licAgreements'),
              icon: 'fas fa-cog'
            },
            {
              href: { name: 'licenses-compliance' },
              title: this.$i18n.t('menu.licCompliance'),
              icon: 'fas fa-cog'
            },
            {
              href: { name: 'licenses-used' },
              title: this.$i18n.t('menu.licUsed'),
              icon: 'fas fa-cog'
            }
          ]
        },
        {
          href: { name: 'cloud-recommendations' },
          title: this.$i18n.t('menu.cloudRec'),
          icon: 'fas fa-cloud'
        }
        // {
        //   href: { name: 'settings' },
        //   title: 'Settings',
        //   icon: 'fas fa-cogs',
        //   child: [
        //     {
        //       title: 'Agents',
        //       icon: 'fas fa-cog'
        //     }
        //   ]
        // }
      ]
    }
  }
}
</script>

<style lang="scss">
$primary-color: #679189 !default;
$base-bg: #363636 !default;
$icon-bg: #363636 !default;
$item-font-size: 14px !default;
$item-line-height: 20px !default;
$item-padding: 15px !default;
$icon-height: 20px !default;
$icon-width: 20px !default;
@import 'vue-sidebar-menu/src/scss/vue-sidebar-menu.scss';

.v-sidebar-menu {
  z-index: 53;

  .vsm--scroll-wrapper {
    margin-right: 0 !important;

    > .vsm--list {
      margin-top: 56px;
    }
  }

  .vsm--toggle-btn {
    position: absolute;
    top: 0;
    left: 0;
    height: 56px;
    width: 50px;

    > span {
      > span {
        margin: 8px 10px;
        float: left;
      }
    }
  }
}
</style>
