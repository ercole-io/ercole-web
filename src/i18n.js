import Vue from 'vue'
import VueI18n from 'vue-i18n'
import getBrowserLocale from '@/util/get-browser-locale.js'
import { supportedLocalesInclude } from '@/util/supported-locales.js'

Vue.use(VueI18n)

function loadLocaleMessages() {
  const locales = require.context(
    './locales',
    true,
    /[A-Za-z0-9-_,\s]+\.json$/i
  )
  const messages = {}
  locales.keys().forEach(key => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i)
    if (matched && matched.length > 1) {
      const locale = matched[1]
      messages[locale] = locales(key)
    }
  })
  return messages
}

function getStartingLocale() {
  const browserLocale = getBrowserLocale({ countryCodeOnly: true })

  if (supportedLocalesInclude(browserLocale)) {
    return browserLocale
  } else {
    return process.env.VUE_APP_I18N_LOCALE || 'en'
  }
}

export default new VueI18n({
  locale: getStartingLocale(),
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true
})
