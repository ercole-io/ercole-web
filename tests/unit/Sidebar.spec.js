import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import Sidebar from '@/components/Sidebar.vue'
import i18n from '@/i18n'

Vue.config.ignoredElements = ['b-icon']

let wrapper
beforeEach(() => {
  wrapper = shallowMount(Sidebar, { i18n })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Sidebar.vue', () => {
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('verify if component exists', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('will emit a collapse value when clicked on button', () => {
    wrapper.vm.onToggleCollapse(true)

    expect(wrapper.emitted().collapsedSidebar[0]).toEqual([true])
  })
})
