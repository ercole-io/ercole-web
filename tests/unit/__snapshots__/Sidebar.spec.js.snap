// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component will mount the component 1`] = `
<div class="v-sidebar-menu vsm_collapsed" style="max-width: 50px;" hiddenoncollapse="true" data-sidebar="">
  <div class="vsm--scroll-wrapper" style="margin-right: -17px;">
    <div class="vsm--list" style="width: 50px;">
      <div class="vsm--item"><a target="_self" href="[object Object]" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-home">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="[object Object]" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-server">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="#" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-database">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="[object Object]" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-object-group">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="[object Object]" class="vsm--link vsm--link_level-1"><i class="vsm--icon far fa-calendar-times">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="[object Object]" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-bell">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
      <div class="vsm--item"><a target="_self" href="#" class="vsm--link vsm--link_level-1"><i class="vsm--icon fas fa-dollar-sign">

          </i>
          <!---->
          <!----></a>
        <!---->
      </div>
    </div>
    <div class="vsm--mobile-item" style="position: absolute; top: 0px; left: 0px; padding-left: 50px; z-index: 0; width: 0px; max-width: 250px;">
      <!---->
      <!---->
    </div>
  </div> <button class="vsm--toggle-btn vsm--toggle-btn_slot"><span><b-icon pack="fas" icon="bars" size="is-small"></b-icon></span></button>
</div>
`;
